packages/site/site-react/src/graphQLFetch/graphQLFetch.ts

```diff
@@ -0,0 +1,123 @@
+type Fetch = typeof fetch;
+
+export type BlockPreviewData = {
+    includeInvisible: boolean;
+};
+
+export function convertPreviewDataToHeaders(previewData?: BlockPreviewData) {
+    const { includeInvisibleBlocks, includeInvisiblePages } = {
+        includeInvisiblePages: !!previewData,
+        includeInvisibleBlocks: previewData && previewData.includeInvisible,
+    };
+
+    const headers: Record<string, string> = {};
+    const includeInvisibleContentHeaderEntries: string[] = [];
+    if (includeInvisiblePages) {
+        includeInvisibleContentHeaderEntries.push("Pages:Unpublished");
+    }
+    if (includeInvisibleBlocks) {
+        includeInvisibleContentHeaderEntries.push("Blocks:Invisible");
+    }
+    // tells api to send invisble content
+    // authentication is required when this header is used
+    if (includeInvisibleContentHeaderEntries.length > 0) {
+        headers["x-include-invisible-content"] = includeInvisibleContentHeaderEntries.join(",");
+    }
+    return headers;
+}
+
+// Adapted from https://github.com/jasonkuhrt/graffle/blob/main/src/layers/6_helpers/gql.ts
+export const gql = (chunks: TemplateStringsArray, ...variables: unknown[]): string => {
+    return chunks.reduce((acc, chunk, index) => {
+        let variable;
+
+        if (index in variables) {
+            if (typeof variables[index] !== "string") {
+                let errorMessage =
+                    `Non-string variable in the GraphQL document\n\n` +
+                    `This is most likely due to importing a GraphQL document from a React Client Component.\n` +
+                    `All GraphQL documents need to be imported from React Server Components (i.e. no "use client" notation).`;
+
+                if (chunk.trim().length > 0) {
+                    errorMessage += `\n\nThe error occurred in the following GraphQL document:\n${chunk}`;
+                }
+
+                throw new Error(errorMessage);
+            } else {
+                variable = variables[index];
+            }
+        } else {
+            variable = "";
+        }
+
+        return `${acc}${chunk}${variable}`;
+    }, ``);
+};
+
+export function createFetchWithPreviewHeaders(fetch: Fetch, previewData?: BlockPreviewData): Fetch {
+    const defaultHeaders = convertPreviewDataToHeaders(previewData);
+    return createFetchWithDefaults(fetch, { headers: defaultHeaders });
+}
```

Comment:
I noticed that this method isn't used anywhere in the Demo. Maybe it's a leftover?

Reply:
it's used in a few of our projects;

however: some (not all) use `createFetchWithPreviewHeaders` PLUS `createFetchWithDefaults` and `convertPreviewDataToHeaders`, which sets the preview headers twice. (won't do any harm, but is unnecessary)

Reply:
Then I'll keep it